[Merged by Bors] - Mutual TLS and S3 TLS verification (#244)
stackabletech/trino-operator: docs/modules/ROOT/pages/usage.adoc

./trino-cli-387-executable.jar --debug --server http://<node>:<http-port> --user=admin
----

==== Secure for production:

There are multiple ways to secure a Trino cluster:

1. Enable Authentication
2. Enable TLS only to connect to the coordinator

> 3. Enable internal TLS for communications between Coordinators and Workers

Author: done

For testing purposes we use the https://trino.io/docs/current/installation/cli.html[Trino CLI].

===== Via Authentication

If authentication is enabled, https://trino.io/docs/current/security/tls.html[TLS] for the coordinator as well as a shared secret for https://trino.io/docs/current/security/internal-communication.html[internal communications] (this is base64 and not TLS encrypted) must be configured.

> Securing the Trino cluster will disable all HTTP port and disable the UI on the HTTP port as well.

Author: done

./trino-cli-387-executable.jar --debug --server http://<node>:<http-port> --user=admin
----

==== Secure for production:

> There are multiple ways to secure a Trino cluster:

Reviewer: ```suggestion
There are multiple steps that must be taken to secure a Trino cluster:
```

1. Enable authentication

> 2. Enable TLS only to connect to the coordinator

Reviewer: ```suggestion
2. Enable TLS between the clients and coordinator
```

3. Enable internal TLS for communications between coordinators and workers

For testing purposes we use the https://trino.io/docs/current/installation/cli.html[Trino CLI].

===== Via authentication

> If authentication is enabled, https://trino.io/docs/current/security/tls.html[TLS] for the coordinator as well as a shared secret for https://trino.io/docs/current/security/internal-communication.html[internal communications] (this is base64 and not TLS encrypted) must be configured.

Reviewer: ```suggestion
If authentication is enabled, https://trino.io/docs/current/security/tls.html[TLS] for the coordinator as well as a shared secret for https://trino.io/docs/current/security/internal-communication.html[internal communications] (this is base64 and not encrypted) must be configured.
```

  name: simple-trino
spec:
  version: 387-stackable0.1.0
  config:
    tls:
      secretClass: trino-tls
  authentication:
    method:
      multiUser:
        userCredentialsSecret:
          name: simple-trino-users-secret
[..]
----

> If no `config.tls.secretClass` is provided but authentication is enabled, it will default to `tls` provided by the https://github.com/stackabletech/secret-operator[Secret Operator].

Author: done
